problem with containsOnlyText()

I have an HTML table; it is quite simple; I find individual records with XPath `"//table/tbody/tr[" + row + "]/td[" + column + "]"`.

When I try to verify the text in the record, I use something like:
```
then(chuck).should(seeThat(the(EventTypesData.item(2, 1)), containsOnlyText("EDIT 2-19")));
```
This all works perfectly fine!

When I click on this record, it changes to an input field with the same text already entered. I find this record with similar XPath: `"//table/tbody/tr[" + row + "]/td[" + column + "]//input"`. And I try the same verification as above:

```
then(chuck).should(seeThat(the(EventTypesData.editableItem(2, 1)), containsOnlyText("EDIT 2-19")));
```

This fails with the following error:
```
java.lang.AssertionError: 
Expected:  an element containing only 'EDIT 2-19'
     but:  was the element identified by xpath: //table/tbody/tr[2]/td[1]//input [EDIT 2-19]
	at features.technical_view.EventTypesEditHistoryStory.should_be_able_to_keep_edit_history(EventTypesEditHistoryStory.java:93)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)
```

I can see my correct XPath in the error. I can see my text in **both** the expected and actual; the square brackets in the error message are added by the framework (in `net.serenitybdd.screenplay.matchers.statematchers.WebElementStateDescription#forElement()` line 9).

Why does this fail? How do I fix it?

--- a/serenity-core/src/main/java/net/serenitybdd/core/pages/WebElementFacadeImpl.java
+++ b/serenity-core/src/main/java/net/serenitybdd/core/pages/WebElementFacadeImpl.java
@@ -605,7 +605,17 @@ public class WebElementFacadeImpl implements WebElementFacade, net.thucydides.co
         }
 
         WebElement element = getElement();
-        return ((element != null) && (element.getText().equals(value)));
+
+	if (element == null)
+	    return false;
+
+	String text = element.getText();
+	if (text.isEmpty()) {
+	    // https://github.com/serenity-bdd/serenity-core/issues/2134
+	    // maybe it's an input element?
+	    text = element.getAttribute("value");
+	}
+	return text.equals(value);
     }
 
     /**

